src/smart-components/workflow/workflows.js

```diff
@@ -24,163 +23,140 @@ const columns = [{
+    const request = () => props.fetchWorkflows(options);
+    if (debounce) {
+      return debouncePromise(request, 250)();
     }
 
-    setOpen = (data, id) => data.map(row => {
-      if (row.id === id) {
-        return {
-          ...row,
-          isOpen: !row.isOpen
-        };
-      }
+    return request().then(({ value: { data }}) => setRows(createInitialRows(data)));
+  };
 
-      return { ...row };
+  const setOpen = (data, id) => data.map(row => row.id === id ?
+    {
+      ...row,
+      isOpen: !row.isOpen
+    } : {
+      ...row
     });
 
-    setSelected = (data, id) => data.map(row => {
-      if (row.id === id) {
-        return {
-          ...row,
-          selected: !row.selected
-        };
-      }
+  const setSelected = (data, id) => data.map(row => row.id === id ?
+    {
+      ...row,
+      selected: !row.selected
+    } : {
+      ...row
+    });
 
-      return { ...row };
-    })
-
-    onFilterChange = filterValue => this.setState({ filterValue })
-
-    onCollapse = (_event, _index, _isOpen, { id }) => this.setState(({ rows }) => ({ rows: this.setOpen(rows, id) }));
-
-    selectRow = (_event, selected, index, { id } = {}) => index === -1
-      ? this.setState(({ rows }) => ({ rows: rows.map(row => ({ ...row, selected })) }))
-      : this.setState(({ rows }) => ({ rows: this.setSelected(rows, id) }));
-
-    renderToolbar() {
-      return (
-        <TableToolbar>
-          <Level style={ { flex: 1 } }>
-            <LevelItem>
-              <Toolbar>
-                <FilterToolbar onFilterChange={ this.onFilterChange } searchValue={ this.state.filterValue } placeholder='Find a Workflow' />
-                <ToolbarGroup>
-                  <ToolbarItem>
-                    <Link to="/workflows/add-workflow">
-                      <Button
-                        variant="primary"
-                        aria-label="Create Workflow"
-                      >
-                Create Workflow
-                      </Button>
-                    </Link>
-                  </ToolbarItem>
-                </ToolbarGroup>
-              </Toolbar>
-            </LevelItem>
-
-            <LevelItem>
-              <Toolbar>
-                <ToolbarGroup>
-                  <ToolbarItem>
-                    <Pagination
-                      itemsPerPage={ this.props.pagination.limit || 50 }
-                      numberOfItems={ this.props.pagination.count || 50 }
-                      onPerPageSelect={ this.handleOnPerPageSelect }
-                      page={ getCurrentPage(this.props.pagination.limit, this.props.pagination.offset) }
-                      onSetPage={ this.handleSetPage }
-                      direction="down"
-                    />
-                  </ToolbarItem>
-                </ToolbarGroup>
-              </Toolbar>
-            </LevelItem>
-          </Level>
-        </TableToolbar>
-      );
+  const onFilterChange = filterValue => setFilterValue(filterValue);
```

Comment:
I would use this directly no point of this function now. You will have to pass only the first value
```
onChange={value => setFiltervalue(value)}
```